src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -184,36 +184,71 @@ void MeshForwarder::HandleResolved(const Ip6::Address &aEid, otError aError)
 
 otError MeshForwarder::EvictMessage(uint8_t aPriority)
 {
-    otError  error = OT_ERROR_NOT_FOUND;
-    Message *message;
+    otError        error    = OT_ERROR_NOT_FOUND;
+    PriorityQueue *queues[] = {&mResolvingQueue, &mSendQueue};
+    Message *      evict    = NULL;
 
-    VerifyOrExit((message = mSendQueue.GetTail()) != NULL);
+    // search for a lower priority message to evict (choose lowest priority message among all queues)
+    for (uint8_t index = 0; index < OT_ARRAY_LENGTH(queues); index++)
+    {
+        for (uint8_t priority = 0; priority < aPriority; priority++)
```

Comment:
nit: small optimization
```suggestion
        for (uint8_t priority = 0; priority < (evict ? evict->GetPriority() : aPriority); priority++)
```

Reply:
@simonlingoogle, great idea! I implemented this optimization in a different way. Please review.